server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4484,9 +4484,32 @@ public Durability getTranslogDurability() {
     // we can not protect with a lock since we "release" on a different thread
     private final AtomicBoolean flushOrRollRunning = new AtomicBoolean();
 
+    /**
+     * Checks if the shard need to be refreshed depending on translog constraints.
+     * each translog type can have it's own decider
+     * @return {@code true} if the shard should be refreshed
+     */
+    public boolean shouldRefreshShard() {
+        final Engine engine = getEngineOrNull();
+        if (engine != null) {
+            try {
+                return engine.translogManager().shouldRefreshShard();
+            } catch (final AlreadyClosedException e) {
```

Comment:
Any reason to not log anything in case of already closed exception?

Reply:
Had discussion on same with @gbbafna . We might not want to raise an exception here, since it's just a new check (not actually doing any task) and exception might already being thrown at other write flow.